bento/geometry/_geometry.py

```diff
@@ -1,310 +1,172 @@
+    SpatialData
+        .points[points_key]: Updated points DataFrame with boolean column for each shape
     """
-    adata = data.copy() if copy else data
 
     if isinstance(shape_names, str):
         shape_names = [shape_names]
 
-    points = get_points(data, points_key, asgeo=True).sort_values("cell")
-    points = points.drop(
-        columns=shape_names, errors="ignore"
-    )  # Drop columns to overwrite
-    points_grouped = points.groupby("cell", observed=True)
-    cells = list(points_grouped.groups.keys())
-    point_sindex = []
-
-    # Iterate over cells and index points to shapes
-    for cell in tqdm(cells, leave=False):
-        pt_group = points_grouped.get_group(cell)
-
-        # Get shapes to index in current cell
-        cur_shapes = gpd.GeoDataFrame(geometry=data.obs.loc[cell, shape_names].T)
-        cur_sindex = (
-            pt_group.reset_index()
-            .sjoin(cur_shapes, how="left", op="intersects")
-            .drop_duplicates(subset="index", keep="first")
-            .sort_index()
-            .reset_index()["index_right"]
-            .astype(str)
-        )
-        point_sindex.append(cur_sindex)
-
-    # TODO: concat is hella slow
-    point_sindex = (
-        pd.concat(point_sindex, ignore_index=True).str.get_dummies() == 1
-    ).fillna(False)
-    point_sindex.columns = [col.replace("_shape", "") for col in point_sindex.columns]
-
-    # Add new columns to points
-    points[point_sindex.columns] = point_sindex.values
-    adata.uns[points_key] = points
-
-    return adata if copy else None
-
-
-def crop(
-    data: AnnData,
-    xlims: Tuple[int],
-    ylims: Tuple[int],
-    copy: bool = True,
-) -> Optional[AnnData]:
-    """Returns a view of data within specified coordinates.
-
-    Parameters
-    ----------
-    data : AnnData
-        Spatial formatted AnnData object
-    xlims : list, optional
-        Upper and lower x limits, by default None
-    ylims : list, optional
-        Upper and lower y limits, by default None
-    copy : bool, optional
-        Whether to return a copy the AnnData object. Default True.
-
-    Returns
-    -------
-    AnnData
-        AnnData object with data cropped to specified coordinates
+    # Grab all shape GeoDataFrames to index points to
+    shape_gpds = {}
+    for shape in shape_names:
+        shape_gpds[shape] = sdata.shapes[shape]
+
+    # Grab points as GeoDataFrame
+    points = sdata.points['transcripts'].compute()
+    points_gpd = gpd.GeoDataFrame(points, geometry=gpd.points_from_xy(points.x, points.y), copy=True)
+
+    # Index points to shapes
+    for shape in shape_gpds:
+        shape_gpd = shape_gpds[shape]
+        sjoined_points = gpd.sjoin(points_gpd, shape_gpd, how="left", predicate="intersects")
+        sjoined_points = sjoined_points[~sjoined_points.index.duplicated(keep='last')]
+        sjoined_points.loc[sjoined_points["index_right"].isna(), "index_right"] = "None"
```

Comment:
should this be set to null instead of a string `None`? as in `None` or `np.nan`

Reply:
Change to empty string instead of "None"